docs/status.md

```diff
@@ -46,7 +46,7 @@ one of the following tracking labels.
 | add                      | yes           | yes          | yes            | yes             | yes         |
 | after_all                | no            | revisit      | no             | yes             | no          |
 | all_gather               | no            | revisit      | no             | no              | no          |
-| all_reduce               | no            | revisit      | revisit        | no              | no          |
+| all_reduce               | yes           | yes          | yes            | no              | no          |
```

Comment:
Where is the type inference?

Reply:
https://github.com/openxla/stablehlo/blob/4aff405bc1936e4121d25d2cd674166275f9c4e1/stablehlo/dialect/StablehloOps.cpp#L291